lib/pagseguro/payment_request.rb

```diff
@@ -82,17 +85,21 @@ def shipping=(shipping)
 
     # Calls the PagSeguro web service and register this request for payment.
     def register
-      path = if @receivers.empty?
-               'checkout'
-             else
-               'checkouts'
-             end
+      request = if @receivers.empty?
+                  Request.post('checkout', api_version, params)
+                else
+                  Request.post_xml('checkouts', api_version, credentials, xml_params)
```

Comment:
Why don't you use only post_xml? The API accepts xml and x-www-form-urlencoded, so if you use just post_xml solve the problem. But this is just a suggest :)